lib/ClangImporter/ClangImporter.cpp

```diff
@@ -631,6 +631,17 @@ importer::getNormalInvocationArguments(
       });
     }
 
+    // To support -apple-none triples, which are non-Darwin, we need to #define
+    // a few things that the Apple SDKs expect.
+    if (triple.getVendor() == llvm::Triple::VendorType::Apple) {
```

Comment:
Shouldn't this be done by the driver and not the clang importer?

Reply:
Hm, that would be against the established pattern -- look at what the rest of this (pretty large) function is doing. It's setting up a ton of things, essentially "construct reasonable defaults for a Clang instance given the current Swift compilation", which my addition seems to match.

I'd say this addition shouldn't go to the driver, on the grounds that that would create an inconsistency. Maybe this whole function should go to the driver? (Even that I would probably argue against.)

Reply:
I'll defer to you, its just a bit odd to splat out the platform knowledge in a bunch of places, cc @artemcm for thoughts (not blocking)

Reply:
I believe that @rauhul is correct here.  `__APPLE__` is a vendor definition, and should always be set by clang for `*-apple-*` triples. I _could_ see the argument for `__MACH__` being the clang importer as _technically_ it is the kernel identifier and not the object file format specifier, but it is used for the object file format as well. I think that sinking this into upstream clang is the right approach.

Reply:
Oh I misunderstood then. I thought that @rauhul was talking about the *Swift driver* and that *it* should receive this logic instead (and pass it as -Xcc flags down to the compiler).